src/main/java/org/elasticsearch/common/settings/ImmutableSettings.java

```diff
@@ -1088,26 +1088,22 @@ public String resolvePlaceholder(String placeholderName) {
                 @Override
                 public boolean shouldIgnoreMissing(String placeholderName) {
                     // if its an explicit env var, we are ok with not having a value for it and treat it as optional
-                    if (placeholderName.startsWith("env.")) {
+                    if (placeholderName.startsWith("env.") || placeholderName.startsWith("prompt.")) {
```

Comment:
Yes we do.

This is still messy for `1.6`. At the time this is called, we haven't prompted the user so the `PlaceholderResolver` isn't able to resolve prompts so they'll always be missing and we need to ignore them. The `shouldRemoveMissingPlaceholder` method is added so that we don't set the value to an empty string; instead we leave the `${prompt.text}` value, which is resolved later on by the `InternalSettingsPreparer`. We also can't add the prompting here without adding another `Map` since a placeholder could be in a system property and we add those to the settings multiple times in the `InternalSettingsPreparer`. 

When we implement #11455, we will need to fix this...